secure/storage/vault/src/lib.rs

```diff
@@ -166,6 +202,75 @@ struct ReadSecretData {
     data: HashMap<String, String>,
 }
 
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+struct ReadPolicyResponse {
+    pub name: String,
+    pub rules: Policy,
+}
+
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+pub struct Policy {
+    #[serde(skip)]
+    internal_rules: PolicyPaths,
+    rules: String,
+}
+
+impl Policy {
+    pub fn new() -> Self {
+        Self {
+            internal_rules: PolicyPaths {
+                path: HashMap::new(),
+            },
+            rules: "".to_string(),
+        }
+    }
+
+    pub fn add_policy(&mut self, path: &str, capabilities: Vec<Capability>) {
+        let path_policy = PathPolicy { capabilities };
+        self.internal_rules
+            .path
+            .insert(path.to_string(), path_policy);
+    }
+
+    pub fn from_json_str(input: &str) -> Result<Self, Error> {
+        let mut policy: Policy = serde_json::from_str(input)?;
+        policy.internal_rules = serde_json::from_str(&policy.rules)?;
+        Ok(policy)
+    }
+
+    pub fn to_json(&self) -> Result<serde_json::Value, Error> {
+        Ok(json!({"rules": serde_json::to_string(&self.internal_rules)?}))
+    }
+}
+
+impl Default for Policy {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[derive(Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct PolicyPaths {
+    path: HashMap<String, PathPolicy>,
+}
+
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+pub struct PathPolicy {
```

Comment:
why the name? it sounds like SecretPolicy and anyway we need a lot of doc to explain those structs and relations between them

Reply:
Paths are more generic than secrets. I admit, I skimped on documentation in this library.